WordPress/gutenberg
# Allow span tag in pagination labels

## What?

Allows `<span>` tags in pagination labels by filter for better flexibility. Addresses [#60878](https://github.com/WordPress/gutenberg/issues/60878). 

## Why?

The current implementation of the `core/query-pagination-previous` and `core/query-pagination-next` blocks in Gutenberg use `esc_html` to escape the label text, which doesn't allow any HTML tags. This limits the freedom to set a label as `Previous <span class="hide-on-mobile">Page</span>`.

## How?

Modifies the `core/query-pagination-previous` and `core/query-pagination-next` blocks to allow `<span>` tags in the pagination labels when filtered.

## Testing Instructions

1. Open a post or page.
2. Insert a Query Loop with Pagination block.
3. Add this to functions.php:
```PHP
function render_block_data_pagination($parsed_block)
{
	if ('core/query-pagination-next' === $parsed_block['blockName']) {
		$parsed_block['attrs']['label'] = 'Next <span class="hide-on-mobile">Page</span>';
	}

	return $parsed_block;
}
add_filter('render_block_data', 'render_block_data_pagination', 10, 3);
```
4. Confirm that pagination navigation next label has the inner span tag.

### Testing Instructions for Keyboard

--
## Screenshots or screencast 
--
